### PR TITLE
Document that apptainer is available via lima

### PR DESCRIPTION
**Pull request by afbjorklund**
## Description of the Pull Request (PR):

Support for Apptainer is already in the main branch of Lima,
and it will be released in the upcoming version of lima ([v0.13](https://github.com/lima-vm/lima/milestone/28))

https://github.com/lima-vm/lima/releases

The main benefit is that it also supports Apple Silicon (ARM).
Also that both Linux and Apptainer are installed automatically.

https://github.com/lima-vm/lima#readme

----

The Windows version is still only in a proof-of-concept phase:
 * https://github.com/lima-vm/lima/issues/909#issuecomment-1192437445

On Mac it uses the HVF acceleration, on Windows it uses WHPX.
It is also possible to run on Linux, where it uses KVM acceleration.

Typical installation:

`brew install lima`
`limactl start template://apptainer`

Then it can be used:

`apptainer`
`apptainer.lima` (on Linux)

Currently the VM is based on Fedora, since it is not in Ubuntu...
You can install the "ubuntu" template and the .deb file, manually.

It is also possible to use Alpine (smaller) or Rocky (bigger) VM,
by installing the `apptainer` package for each Linux distribution.

### CODE REVIEW

**Comment by afbjorklund** [comment]
I thought the old docs were a bit too wordy, so just added a short-short version.

https://github.com/apptainer/singularity-admindocs/blob/master/installation.rst

File: installation.rst [hunk 1]
```diff
@@ -498,3 +514,44 @@ To use Vagrant via Homebrew:
 
    $ /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
    $ brew install --cask virtualbox vagrant vagrant-manager
+
+Here is an example of a Vagrantfile, to be saved (or use ``vagrant init``):
```
Comment (DrDaveD) [inline installation.rst:518]: I don't think 'to be saved' is clear.  I'm not sure exactly what you're trying to say so it's hard for me to suggest an alternate.

File: installation.rst [hunk 2]
```diff
@@ -498,3 +514,44 @@ To use Vagrant via Homebrew:
...
+To use Lima via Homebrew:
+
+.. code::
+
+   $ /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+   $ brew install qemu lima
+
+Then do ``limactl start template://apptainer`` and ``limactl shell apptainer``.
+
+See https://github.com/lima-vm/lima/blob/master/examples/apptainer.yaml
```
Comment (DrDaveD) [inline installation.rst:557]: Let's use a link here instead of a URL.

File: installation.rst [hunk 3]
```diff
@@ -476,19 +476,35 @@ source tools, by following the instructions below. Then you can install
...
 -  `Git for Windows <https://git-for-windows.github.io/>`_
 -  `VirtualBox for Windows <https://www.virtualbox.org/wiki/Downloads>`_
 -  `Vagrant for Windows <https://www.vagrantup.com/downloads.html>`_
 -  `Vagrant Manager for Windows <http://vagrantmanager.com/downloads/>`_
 
+Lima
+----
+
+The Windows version of Lima is still in development.
+You can use `WSL <https://docs.microsoft.com/en-us/windows/wsl/>`_.
```
Comment (DrDaveD) [inline installation.rst:500]: Can you give any hints on what to do in WSL?

**Review by DrDaveD — CHANGES_REQUESTED** [review]

File: installation.rst (hunk 1 above)
Comment (afbjorklund) [reply installation.rst:518]: I was trying to say "save this block of text into a file called Vagrantfile"

File: installation.rst (hunk 3 above)
Comment (afbjorklund) [reply installation.rst:500]: You pick a "distribution", and then follow the Linux instructions

https://learn.microsoft.com/en-us/windows/wsl/basic-commands

Will need to verify on a Windows machine, though.

If I remember correctly, the default OS is Ubuntu ?

File: installation.rst (hunk 1 above)
Comment (afbjorklund) [reply installation.rst:518]: Maybe it _should_ use a separate repository, only to keep the OS version out of the docs ?

Like https://github.com/singularityware/singularity-vagrant/blob/master/singularity/Vagrantfile

The old repository was retired due to the existance of the sylabs "boxes", and they _they_ retired...

Now that there are binary packages, it should be as simple as copy/paste of the Linux instructions ?

**Comment by afbjorklund** [comment]
I think the PR side-tracked from the original intent (i.e. to add Lima)

But clearing up the situation for Vagrant and for WSL is a _good_ thing...

File: installation.rst [hunk 4]
```diff
@@ -498,3 +521,45 @@ To use Vagrant via Homebrew:
...
+To use Lima via Homebrew:
+
+.. code::
+
+   $ /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+   $ brew install qemu lima
+
+Then do ``limactl start template://apptainer`` and ``limactl shell apptainer``.
+
+The apptainer template for lima looks like this: `examples/apptainer.yaml <https://github.com/lima-vm/lima/blob/master/examples/apptainer.yaml>`_.
```
Comment (DrDaveD) [inline installation.rst:565]: I suggest instead
```
See the `lima apptainer template <https://github.com/lima-vm/lima/blob/master/examples/apptainer.yaml>`_ for more details.
```

**Review by DrDaveD — CHANGES_REQUESTED** [review]

**Comment by afbjorklund** [comment]
BTW; I confirmed that using apptainer 1.1.2 worked fine in Ubuntu 22.04 on WSL2 (Windows 10)

**Review by DrDaveD — APPROVED** [review]
Thanks a lot for the contribution.

**Comment by afbjorklund** [comment]
Note: I'm saving the discussion on how to handle Windows paths and macOS paths for another time.

The current setup assumes that you want to get a Linux shell - and then take it from there...

A future improvement would be to describe how you can stay in your own OS - but run `apptainer`.

i.e. how to get `wsl apptainer` or `lima apptainer` to work, with all of the path mapping etc.
